### PR TITLE
Update to new ESLint class from eslint 7

### PR DESCRIPTION
**Pull request by rachel-fenichel**
This is a clumsy attempt at moving to the new async linter and I need some help with promises.

### CODE REVIEW

**Review by samelhusseini — APPROVED** [review]

**Comment by samelhusseini** [comment]
Your use of promises and async/await looks fine to me. Was there something specific you needed help with?

File: plugins/dev-scripts/scripts/lint.js [hunk 1]
```diff
@@ -25,21 +26,39 @@ console.log(`Running lint for ${packageJson.name}`);
 
 // Create the eslint engine.
 const eslintConfig = require('@blockly/eslint-config');
-const cli = new CLIEngine({
+const cli = new ESLint({
```
Comment (alschmiedt) [inline plugins/dev-scripts/scripts/lint.js]: Should be updated to eslint

Comment (samelhusseini) [reply plugins/dev-scripts/scripts/lint.js]: What do you mean, the variable above is named `ESLint`?

Comment (alschmiedt) [reply plugins/dev-scripts/scripts/lint.js]: I think cli should be eslint.  We are not using CLIEngine anymore so I don't think the name cli makes sense.

Comment (samelhusseini) [reply plugins/dev-scripts/scripts/lint.js]: Oh ok 👍

**Comment by rachel-fenichel** [comment]
Re promises, I thought there's a better way to show that the task failed than to return null and parse it in the caller.

But if it's all good, that works too. This was just a step toward getting lint --fix to be available at the top level.

**Comment by samelhusseini** [comment]
Returning null with async/await, is the equivalent of returning Promise.resolve(null) and you can do that if you like for clarity.

As for errors, the async/await way to return errors is to throw an Error. Although I wouldn't call the directory not existing as an error per se.

**Pull request merged**